### DISCUSSION TITLE
Release notes nothing inside

### DISCUSSION THREAD

**Discussion by DinTri** [rfennell/AzurePipelines#1295] [Q&A] [answered]
Hi,
I tried release notes extension but here is what I receive as a result. I used the template you have provided.
What am I missing?

Here is the result:
🚀 Release v34527 (2.4.0.33980)
Associated Pull Requests (0) ------> ZERO, NULL, not what I want to see
Associated Azure Board Items (0) ----> ZERO, NULL, not what I want to see
Assets
[Release v34527 (2.4.0.33980) as ZIP](https://xxx/_apis/git/repositories/$//items?path=/&versionDescriptor%AZP255BversionOptions%AZP255D=0&versionDescriptor%AZP255BversionType%AZP255D=1&versionDescriptor%AZP255Bversion%AZP255D=v34527 (2.4.0.33980)&resolveLfs=true&%AZP2524format=zip&api-version=5.0&download=true)

Can you please help me?

**Comment by rfennell** [comment, marked as answer]
The associated PRs are found using the merge commit IDs of the PRs, matching these to the commits associated with build. If these are no linked then you will get no matches

A few options to explore

- Run your pipeline with the variable `system.debug=true` or check the 'run with diagnostics' check box when you queue the build
- Use the [local run tester](https://github.com/rfennell/AzurePipelines/tree/main/Extensions/XplatGenerateReleaseNotes/V3/testconsole) - useful for re-running release notes generation off the same build repeatedly

Both of these options will give detailed logging so you can see all the processes to build the list.

**Reply by DinTri** [reply]
Just one more thing to ask. Is it possible when I generate release notes and click redeploy to see them again. Because if I do that the release notes are empty.

**Reply by rfennell** [reply]
It is best to avoid redeployment as the results can differ (as the API calls are based on comparing against last successful build)

Better to put the release not is a separate stage, or write then to a WIKI or similar to avoid the need to regenerate.

This video may help https://www.youtube.com/embed/xaV3dFoQdV8?start=69
